File: lesson6/c_c++/homework/homework6_11.cpp
// lesson6/c_c++/homework/homework6_11.cpp
#include <iostream>

int perfect_number(int numb) {
    int i;
    int res = 0;
    bool res_1;
    for (i = 1; i < numb; i++) {
        if (numb % i == 0) {
            res += i;
        }
        if (numb == res) {
            res_1 = true;
        } else {
            res_1 = false;
        }
    }
    return res_1;
}

int main() {
    int numb = 28;
    int result;
    result = perfect_number(numb);
    if (result){
        std::cout << "совершенное число -> " << numb << '\n';
    } else {
        std::cout << "несовершенное число -> " << numb << '\n';
    }
    return 0;
}

File: lesson6/python3/homework/homework6_11.py
#!/usr/bin/python3
# lesson6/python3/homework/homework6_11.py


def perfect_number(numb):
    res_1 = 0
    res_2 = True
    for i in range(1, numb):
        if numb % i == 0:
            res_1 += i

        if numb == res_1:
            res_2 = True
        else:
            res_2 = False
    return res_2

numb_n = 28
result = perfect_number(numb_n)
if result:
    print("совершенное число ->", numb_n)
else:
    print("несовершенное число ->", numb_n)

